pymeasure/adapters/ni_gpib_232ct.py

```diff
@@ -0,0 +1,328 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2023 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import logging
+from enum import IntFlag, Flag
+import time
+from pymeasure.adapters import VISAAdapter
+from pymeasure.instruments.validators import strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+# pylint: disable=W1203
+
+
+class NI_GPIB_232(VISAAdapter):
+    """Encapsulates the additional commands necessary to with GPIB device communicate over\
+    a National Instruments GPIB-232CT adapter.
+
+    Each adapter is constructed based on a connection to the device
+    itself and the GPIB address of the instrument to be communicated with.
+    Connection sharing is achieved by using the `.gpib` method
+    to spawn new NI_GPIB_232s for different GPIB addresses.
+
+    :param resource_name: A VISA resource string that identifies the\
+    connection to the device itself, for example "ASRL5" for the 5th COM port.
+    :param address: Integer GPIB address of the desired instrument.
+    :param serial_timeout: Timeout value for the serial communication, unit: ms
+    :param rw_delay: internal delay, unit: ms, default 50 ms
```

Comment:
"Internal delay" for what? When is that important? Adding this info makes the parameter more useful and gives needed context for users changing it.

Reply:
So, we normally implement this delay between read and write on the instrument level as a `query_delay` parameter, e.g.  https://pymeasure.readthedocs.io/en/latest/api/instruments/instruments.html#pymeasure.instruments.common_base.CommonBase.ask
(unfortunately in seconds :cry: )

I'm not sure we should duplicate this feature at different levels of the API. 
Is this delay needed for a specific instrument you are using, or is this _always_ needed in this adapter device?

Reply:
this delay seems always needed for the communication with the GPIB devices I have here accessible to me right now

Reply:
@bmoneke what do you think, how should we optimally deal with this? Also name the kwarg `query_delay` to avoid having lots of similar sounding delay names floating around? Intentionally choose a different one? We intentionally deprecated an adapter-wide query_delay in VISAAdapter https://pymeasure.readthedocs.io/en/latest/api/adapters.html#pymeasure.adapters.VISAAdapter ...